be/src/exec/spill/mem_table.cpp

```diff
@@ -47,25 +98,38 @@ Status UnorderedMemTable::append_selective(const Chunk& src, const uint32_t* ind
     Chunk* current = _chunks.back().get();
     size_t mem_usage = current->memory_usage();
     current->append_selective(src, indexes, from, size);
+    _num_rows += size;
     mem_usage = current->memory_usage() - mem_usage;
 
     _tracker->consume(mem_usage);
     COUNTER_ADD(_spiller->metrics().mem_table_peak_memory_usage, mem_usage);
     return Status::OK();
 }
 
-Status UnorderedMemTable::flush(FlushCallBack callback) {
-    while (_processed_index < _chunks.size()) {
-        RETURN_IF_ERROR(callback(_chunks[_processed_index++]));
+Status UnorderedMemTable::finalize() {
+    if (_block == nullptr) {
```

Comment:
We need to serialize into multiple memories instead of one, otherwise it would break the design of yield.